huntsman/dome/musca.py

```diff
@@ -120,10 +123,31 @@ def open(self):
 
         v = self.status()[Protocol.SHUTTER]
         if v == Protocol.OPEN:
+            # refresh the threading close event
+            self._close_event.clear()
+            self.logger.info('Starting thread to keep dome open.')
+            # start a thread to send the keep dome open command to musca
+            keep_open = threading.Thread(target=self.keep_dome_open)
             return True
         self.logger.warning('HuntsmanDome.open wrong final state: {!r}', v)
         return False
 
+    def keep_dome_open(self):
+        """Periodically tell musca to reset watchdog timer
+
+        """
+        # check to see if a dome closure has occured
+        if self._close_event().is_set():
+            self.logger.info('Keep dome open thread has detected a dome closure, ending thread.')
+            # if dome has closed, don't try to 'keep dome open'
+            return
+        self.logger.info('Sending keep dome open command to musca.')
+        self._write_musca(Protocol.KEEP_DOME_OPEN, 'Keeping dome open.')
+        # wait slightly less than 5 minutes to make sure we reset the
+        # dome closure timer before it initiates a dome closure
+        self.logger.debug('Keep dome open thread sleeping for ~5 minutes.')
+        time.sleep(290)
+
```

Comment:
All of this needs to be in a `while True:` loop otherwise it will only send the keep dome open command once.